File: 搜索/深搜/leetcode872/872.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    bool leafSimilar(TreeNode* root1, TreeNode* root2) {
        vector<int> seq1, seq2;
        get_seq(root1, seq1);
        get_seq(root2, seq2);
        return seq1 == seq2 ? true : false;
    }
    
    void get_seq(TreeNode* root, vector<int> &seq) {
        if (!root)
            return;
        if (root->left != NULL)
            get_seq(root->left, seq);
        if (root->right != NULL)
            get_seq(root->right, seq);
        if (!root->left && !root->right)
            seq.push_back(root->val);
    }
};

File: 搜索/深搜/leetcode872/872.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution(object):
    def leafSimilar(self, root1, root2):
        """
        :type root1: TreeNode
        :type root2: TreeNode
        :rtype: bool
        """
        seq1 = []
        seq2 = []
        self.get_seq(root1, seq1)
        self.get_seq(root2, seq2)
        return seq1 == seq2
    
    
    def get_seq(self, root, seq):
        if root == None:
            return
        if root.left != None:
            self.get_seq(root.left, seq)
        if root.right != None:
            self.get_seq(root.right, seq)
        if root.left == None and root.right == None:
            seq.append(root.val)
            